### ISSUE TITLE
writevtk for arbitrary order reference FEs

### ISSUE THREAD

**Issue by fverdugo** [gridap/Gridap.jl#123]
The functionality in order to generate vtu files currently supports 1st and 2nd order reference FEs, but we want to extend it to arbitrary order.

```julia
julia> using Gridap.ReferenceFEs

julia> using Gridap.Visualization

julia> using Gridap.Geometry

julia> reffe = LagrangianRefFE(Float64,QUAD,1)
LagrangianRefFE

julia> grid = Grid(reffe);

julia> writevtk(grid,"grid") # Works!
1-element Array{String,1}:
 "grid.vtu"

julia> reffe = LagrangianRefFE(Float64,QUAD,2)
LagrangianRefFE

julia> grid = Grid(reffe);

julia> writevtk(grid,"grid") # Works!
1-element Array{String,1}:
 "grid.vtu"

julia> reffe = LagrangianRefFE(Float64,QUAD,3)
LagrangianRefFE

julia> grid = Grid(reffe);

julia> writevtk(grid,"grid")
ERROR: This function is not yet implemented
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at /home/fverdugo/Code/jl/Gridap.jl/src/new/Helpers/Helpers.jl:38 [inlined]
 [3] _vtkinfo_extrusion_polytope at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:269 [inlined]
 [4] get_vtkid(::ExtrusionPolytope{2}, ::Gridap.Polynomials.MonomialBasis{2,Float64}) at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:203
 [5] get_vtkid(::LagrangianRefFE{2}) at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:190
 [6] iterate at ./generator.jl:47 [inlined]
 [7] _collect(::Array{LagrangianRefFE{2},1}, ::Base.Generator{Array{LagrangianRefFE{2},1},typeof(Gridap.Visualization.get_vtkid)}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:635
 [8] collect_similar at ./array.jl:564 [inlined]
 [9] map at ./abstractarray.jl:2073 [inlined]
 [10] _vtkcells at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:123 [inlined]
 [11] #write_vtk_file#2(::Dict{Any,Any}, ::Dict{Any,Any}, ::typeof(write_vtk_file), ::Gridap.Visualization.VisualizationGrid{2,2,Float64}, ::String) at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:99
 [12] (::Gridap.Visualization.var"#kw##write_vtk_file")(::NamedTuple{(:celldata, :nodaldata),Tuple{Dict{Any,Any},Dict{Any,Any}}}, ::typeof(write_vtk_file), ::Gridap.Visualization.VisualizationGrid{2,2,Float64}, ::String) at ./none:0
 [13] #writevtk#3(::Int64, ::Int64, ::Dict{Any,Any}, ::Dict{Any,Any}, ::typeof(writevtk), ::UnstructuredGrid{2,2,Float64,false}, ::String) at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:365
 [14] writevtk(::UnstructuredGrid{2,2,Float64,false}, ::String) at /home/fverdugo/Code/jl/Gridap.jl/src/new/Visualization/Vtk.jl:340
 [15] top-level scope at REPL[13]:1
```

**Comment by victorsndvg** [comment]
Ok, I understand the request. I will take a look to understand the code and we will comment how to solve it. Do you agree?

**Comment by fverdugo** [comment]
I add the discussion tag, since it is not clear how to accomplish this.

**Comment by victorsndvg** [comment]
I've been reading a little about it. As we want to visualize high order elements, I took Paraview as starting point to try to discover how to write down and then represent this kind of data.

Is important to highlight that Paraview team tell that [Arbitrary order lagrange elements](https://www.paraview.org/Wiki/ParaView_Release_Notes#Arbitrary-order_Lagrange_finite_element_support) are supported since Paraview version 5.5.0. Anyway, it seems that the documentation about this kind of elements and how to write them to a VTK file is missing.

Some other tools like Deal II or MFEM have already worked on this. I've gather the most relevant info I found. The following links contain some useful discussions and more links:
  - https://github.com/dealii/dealii/issues/6931
  - https://github.com/mfem/mfem/pull/975

It seems that we have to dive into the VTK source code to discover how this elements are written (in particular its connectivities). I also found a project ([VTK_Lagrange](https://github.com/agrayver/VTK_Lagrange)) that can be useful to write some example data.

Finally, I did not check this examples yet. I've to download some of the latest Paraview releases and check if it works.
